cnf-tests/testsuites/e2esuite/multinetworkpolicy/multinetworkpolicy_sriov.go

```diff
@@ -0,0 +1,750 @@
+						Port:     &port6666,
+						Protocol: &protoUDP,
+					}},
+				}),
+				CreateInNamespace(nsX),
+			)
+
+			// Allowed
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, OnPort(port6666), ViaUDP))
+
+			// Not allowed
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port5555), ViaTCP))
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port5555), ViaUDP))
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port6666), ViaTCP))
+		})
+	})
+
+})
+
+func cleanPodsInNamespaces(argNamespaces ...string) {
+	for _, ns := range argNamespaces {
+		Expect(
+			namespaces.CleanPods(ns, client.Client)).
+			ToNot(HaveOccurred())
+	}
+}
+
+func cleanMultiNetworkPolicies(namespace string) {
+	err := client.Client.MultiNetworkPolicies(namespace).
+		DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(func() int {
+		ret, err := client.Client.MultiNetworkPolicies(namespace).
+			List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return len(ret.Items)
+	}, 30*time.Second, 1*time.Second).Should(BeZero())
+}
+
+func createNamespaces(argNamespaces ...string) {
+	for _, ns := range argNamespaces {
+		Expect(
+			namespaces.Create(ns, client.Client),
+		).ToNot(HaveOccurred())
+	}
+}
+
+func createPodsInNamespace(namespace string) (*corev1.Pod, *corev1.Pod, *corev1.Pod) {
+
+	podA := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podA, "pod", "a")
+	pods.RedefinePodWithNetwork(podA, TEST_NETWORK)
+	redefineWithNetcatServers(podA)
+	_, err := pods.RedefineAsPrivileged(podA, "")
+	Expect(err).To(BeNil())
+	podA.ObjectMeta.GenerateName = "testpod-a-"
+	podA = createAndStartPod(podA)
+
+	podB := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podB, "pod", "b")
+	pods.RedefinePodWithNetwork(podB, TEST_NETWORK)
+	redefineWithNetcatServers(podB)
+	_, err = pods.RedefineAsPrivileged(podB, "")
+	Expect(err).To(BeNil())
+	podB.ObjectMeta.GenerateName = "testpod-b-"
+	podB = createAndStartPod(podB)
+
+	podC := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podC, "pod", "c")
+	pods.RedefinePodWithNetwork(podC, TEST_NETWORK)
+	redefineWithNetcatServers(podC)
+	_, err = pods.RedefineAsPrivileged(podC, "")
+	Expect(err).To(BeNil())
+	podC.ObjectMeta.GenerateName = "testpod-c-"
+	podC = createAndStartPod(podC)
+
+	return podA, podB, podC
+}
+
+func waitForObject(clientSet *sriovtestclient.ClientSet, key runtimeclient.ObjectKey, object runtimeclient.Object) {
+
+	Eventually(func() error {
+		return clientSet.Get(context.Background(), key, object)
+	}, 120*time.Second, 1*time.Second).
+		WithOffset(1).
+		ShouldNot(
+			HaveOccurred(),
+			"Object [%s] not found for key [%s]", reflect.TypeOf(object), key,
+		)
+}
+
+// Add 4 containers to the pod for netcat server. 2 for each transport protocol (TCP/UDP) for each
+// port in (5555/6666)
+func redefineWithNetcatServers(pod *corev1.Pod) *corev1.Pod {
```

Comment:
is this function in use?

Reply:
Yes,  in `createPodsInNamespace(...)`

https://github.com/openshift-kni/cnf-features-deploy/pull/1113/files#diff-b96c3f1d12439337ea75489fadce58f869013344d564ecbb6731fc0ba93a5f67R694

As netcat servers installed by this function are intended to be used by the ReachMatcher, I'll put it in `reachmatcher.go` file.